bench: refactor random number generation in JS benchmarks for `stats/base/dists/uniform`

G4URAV001 opened the pull request:
…ts/uniform

#4991

Resolves #4991 .

## Description

> What is the purpose of this pull request?

This pull request improves random number generation in JS benchmarks for `stats/base/dists/uniform`. Specifically:

- Moves random number generation out of benchmarking loops to prevent interference with benchmark results.
- Uses `@stdlib/random/base/uniform` and `@stdlib/random/base/discrete-uniform` instead of `randu` expressions.
- Ensures generated random values maintain the same range as existing values for consistency.

## Related Issues

>This pull request:

- Resolves #4991
- References #4837 and #4955 as examples of similar improvements.


## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

   ✅ Read, understood, and followed the [contributing guidelines][contributing].
   ✅ Build Native Add-on.
   ✅ Run JavaScript Benchmarks.
   ✅ Run JavaScript Native Benchmarks.



* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

G4URAV001 committed:
Refactor random number generation in JS benchmarks for stats/base/dists/uniform
#4991

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/cdf/benchmark/benchmark.js:
```diff
@@ -32,16 +33,24 @@ var cdf = require( './../lib' );
+	min = new Float64Array(len);
+	max = new Float64Array(len);
+	for(i = 0; i < len; i++){
+		x[ i ] = uniform(-10.0, 20.0);
+		min[ i ] = uniform(-20.0, 20.0);
```
```suggestion
		x[ i ] = uniform( -10.0, 10.0 );
		min[ i ] = uniform( -20.0, 0.0 );
```

The range remains the same as in the previous code. Also, brackets should have spaces at the beginning and end.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/cdf/benchmark/benchmark.js:
```diff
@@ -32,16 +33,24 @@ var cdf = require( './../lib' );
+	for(i = 0; i < len; i++){
+		x[ i ] = uniform(-10.0, 20.0);
+		min[ i ] = uniform(-20.0, 20.0);
+		max[ i ] = min[ i ] + uniform(0.0, 40.0);
```
```suggestion
		max[ i ] = uniform( min[ i ], min[ i ] + 40.0 );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/cdf/benchmark/benchmark.js:
```diff
@@ -32,16 +33,24 @@ var cdf = require( './../lib' );
-		min = ( randu() * 20.0 ) - 20.0;
-		max = min + ( randu() * 40.0 );
-		y = cdf( x, min, max );
+		y = cdf( x[ i % len], min[ i % len], max[ i % len] );
```
```suggestion
		y = cdf( x[ i % len ], min[ i % len ], max[ i % len ] );
```

This applies throughout the PR

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/cdf/benchmark/benchmark.js:
```diff
@@ -58,18 +67,23 @@ bench( pkg+':factory', function benchmark( b ) {
+	len = 100;
+	x = new Float64Array(len);
+	for(i = 0; i < len; i++){
+		x[ i ] = uniform(-2.0, 2.0);
```
```suggestion
		x[ i ] = uniform( -2.0, 0.0 );
```

Same comments as above

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/cdf/benchmark/benchmark.js:
```diff
@@ -58,18 +67,23 @@ bench( pkg+':factory', function benchmark( b ) {
 	for ( i = 0; i < b.iterations; i++ ) {
-		x = ( randu()*2.0 ) - 2.0;
-		y = mycdf( x );
+		y = mycdf( x[i % len] );
```
```suggestion
		y = mycdf( x[ i % len ] );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/ctor/benchmark/benchmark.js:
```diff
@@ -32,15 +33,22 @@ var Uniform = require( './../lib' );
+	len = 100;
+	a = new Float64Array(len);
+	b = new Float64Array(len);
+	for(i = 0; i < len; i++){
+		a[ i ] = uniform(EPS, 10.0);
+		b[ i ] = uniform(EPS, 10.0);
+	}
```
```suggestion
	for( i = 0; i < len; i++ ) {
		a[ i ] = uniform( EPS, 10.0 );
		b[ i ] = uniform( EPS, 10.0 );
	}
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/ctor/benchmark/benchmark.js:
```diff
@@ -142,17 +156,21 @@ bench( pkg+'::set:b', function benchmark( bm ) {
 	dist = new Uniform( a, b );
+	len = 100;
+	y = new Float64Array(len);
+	for(i = 0; i < len; i++){
+		y[ i ] = uniform(a + EPS, 100.0)
+	}
```
Same comment as above and applies throughout the PR.

anandkaranubc commented:
Thanks @G4URAV001 for working on this. Left a few comments for you to review. Good luck!

G4URAV001 committed:
improved on given comments issue #4991

G4URAV001 committed:
issue #4991 all test done

G4URAV001 commented:
> Thanks @G4URAV001 for working on this. Left a few comments for you to review. Good luck!

Thanks @anandkaranubc for the comments. Did all the correction and ran the following tests locally in a dev container - 
   
[x] Build Native Add-on.
   [x] Run JavaScript Benchmarks.
   [x] Run JavaScript Native Benchmarks.

Still if anything is missed I can do the changes

Sorry for the silly things I missed earlier, I should have been more observant.

Looking forward for more contributions

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/stdev/benchmark/benchmark.js:
```diff
@@ -32,14 +32,22 @@ var stdev = require( './../lib' );
+	len = 100;
+	min = new Float64Array( len );
+	max = new Float64Array( len );
+	for( i = 0; i < len; i++ ){
```
```suggestion
	for ( i = 0; i < len; i++ ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/skewness/benchmark/benchmark.js:
```diff
@@ -32,14 +32,21 @@ var skewness = require( './../lib' );
+	len = 100;
+	min = new Float64Array( len );
+	max = new Float64Array( len );
+	for( i = 0; i < len; i++ ){
```
```suggestion
	for ( i = 0; i < len; i++ ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/quantile/benchmark/benchmark.js:
```diff
@@ -58,18 +66,23 @@ bench( pkg+':factory', function benchmark( b ) {
 	myquantile = quantile.factory( min, max );
+	len = 100;
+	p = new Float64Array( len );
+	for(i = 0; i < len; i++){
```
```suggestion
	for ( i = 0; i < len; i++ ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/quantile/benchmark/benchmark.js:
```diff
@@ -32,16 +32,24 @@ var quantile = require( './../lib' );
+	p = new Float64Array( len );
+	min = new Float64Array( len );
+	max = new Float64Array( len );
+	for( i = 0; i < len; i++ ){
```
```suggestion
	for ( i = 0; i < len; i++ ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/median/benchmark/benchmark.js:
```diff
@@ -21,7 +21,7 @@
 
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require('@stdlib/random/base/uniform');
```
```suggestion
var uniform = require( '@stdlib/random/base/uniform' );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/median/benchmark/benchmark.native.js:
```diff
@@ -23,7 +23,7 @@
 var bench = require( '@stdlib/bench' );
 var Float64Array = require( '@stdlib/array/float64' );
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require('@stdlib/random/base/uniform');
```
```suggestion
var uniform = require( '@stdlib/random/base/uniform' );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/benchmark/benchmark.js:
```diff
@@ -22,10 +22,10 @@
 var bench = require( '@stdlib/bench' );
 var Float64Array = require( '@stdlib/array/float64' );
-var randu = require( '@stdlib/random/base/randu' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pkg = require( './../package.json' ).name;
 var entropy = require( './../lib' );
+var uniform = require('@stdlib/random/base/uniform');
```
```suggestion
var uniform = require( '@stdlib/random/base/uniform' );
var isnan = require( '@stdlib/math/base/assert/is-nan' );
var pkg = require( './../package.json' ).name;
var entropy = require( './../lib' );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/benchmark/benchmark.native.js:
```diff
@@ -24,7 +24,7 @@ var resolve = require( 'path' ).resolve;
 var Float64Array = require( '@stdlib/array/float64' );
 var tryRequire = require( '@stdlib/utils/try-require' );
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require('@stdlib/random/base/uniform');
```
```suggestion
var uniform = require( '@stdlib/random/base/uniform' );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/kurtosis/benchmark/benchmark.js:
```diff
@@ -22,7 +22,7 @@
 var bench = require( '@stdlib/bench' );
 var Float64Array = require( '@stdlib/array/float64' );
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require('@stdlib/random/base/uniform');
```
```suggestion
var uniform = require( '@stdlib/random/base/uniform' );
```

Planeshifter commented:
/stdlib lint-autofix

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/mgf/benchmark/benchmark.js:
```diff
@@ -21,7 +21,7 @@
 
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require('@stdlib/random/base/uniform');
```
```suggestion
var uniform = require( '@stdlib/random/base/uniform' );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/mgf/benchmark/benchmark.js:
```diff
@@ -32,16 +32,24 @@ var mgf = require( './../lib' );
+	t = new Float64Array( len );
+	min = new Float64Array( len );
+	max = new Float64Array( len );
+	for( i = 0; i < len; i++ ){
```
```suggestion
	for ( i = 0; i < len; i++ ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/mgf/benchmark/benchmark.js:
```diff
@@ -58,18 +66,23 @@ bench( pkg+':factory', function benchmark( b ) {
 	mymgf = mgf.factory( min, max );
+	len = 100;
+	t = new Float64Array( len );
+	for( i = 0; i < len; i++ ){
```
```suggestion
	for ( i = 0; i < len; i++ ) {
```

stdlib-bot committed:
fix: resolve lint errors

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/logcdf/benchmark/benchmark.js:
```diff
@@ -21,7 +21,7 @@
 
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require( '@stdlib/random/base/uniform' );
```
```suggestion
var Float64Array = require( '@stdlib/array/float64' );
var uniform = require( '@stdlib/random/base/uniform' );
```

Planeshifter committed:
bench: add missing dependency

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

Planeshifter committed:
chore: resolve lint errors

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter committed:
chore: resolve lint errors

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter reviewed (approved):
Thanks for your first PR, which I will go ahead and merge shortly.

Going forward, please make sure you have your local development environment setup correctly, including EditorConfig and ESLint. The PR had a bunch of lint errors that would have been caught by our GIt hooks if properly initialized.

Planeshifter committed:
chore: fix lint errors
